mkosi

```diff
@@ -97,10 +97,10 @@ class CommandLineArguments(argparse.Namespace):
     esp_partno: Optional[int] = None
     xbootldr_partno: Optional[int] = None
 
-    def generated_root(self):
+    def generated_root(self) -> bool:
         """Returns whether this configuration means we need to generate a file system from a prepared tree,
         as needed for anything squashfs and when root minimization is required."""
-        return self.minimize or self.output_format.is_squashfs()
+        return cast(bool, self.minimize) or self.output_format.is_squashfs()
```

Comment:
Shouldn't there be a line like
```python3
minimize: bool = False
```
in the declaration section instead? Having to do a cast sounds wrong.

Reply:
hmm, this is something https://github.com/poettering/mkosi/pull/1 added and I just merged. My python-fu is limited. @behrmann can you elaborate?

Reply:
I'm not too happy about the cast myself and would love a cleaner solution. If I remember correctly, `self.minimize` comes directly from argparse via the `BooleanAction` and I haven't yet found the proper incantations to have mypy figure out that the value must hence be boolean. I'll have a look whether mypy can be persuaded to do that.